test-suite/tests/compact-p002-out.jsonld

```diff
@@ -0,0 +1,8 @@
+{
+  "@context": {
+    "ex": {"@id": "http://example.org/"}
+  },
+  "@id": "http://example.org/id1",
+  "@type": ["http://example.org/Type1", "http://example.org/Type2"],
+  "http://example.org/term": {"@id": "http://example.org/id2"}
+}
```

Comment:
@niklasl `ex` is not used, because it's not a simple term, this test is there to be sure that a term with an expanded term definition is not used for creating a compact IRI. Certainly, when expanding, it would be used, but most expanded term definitions are defined for property values, not things intended as prefixes, this the problem with defining a term "resource", and finding that when compacting, you get "resource:s", which the limitation is attempting to address.

I originally did use 1.1 processing rules for this, and we could certainly put that back in, but my worry was that 1.0 processors will continue to do things differently, and this would be confusing in any case. Also, `ex:` can certainly be used in 1.0 to create `ex::term`, which is allowed, but odd. This just fixes that behavior.

The theory is that compact IRI generation in 1.0 is basically broken, so we should attempt to fix it without invoking the heavy-handed step of needing to explicitly define this as a 1.1 processing mode.